### PR TITLE
core:unwind: check user context on stack print of panicked TAs

### PR DESCRIPTION
**Pull request by etienne-lms**
This change checks that the userland context pointer is valid before
reading its content.

Note that this change only lowers the chance of malformed TA being
able to crash core or access core memory using crafted context
reference. The stack unwind process being executed from kernel land,
a real fix could require each stack unwind step to verify the memory
references before going further in the execution history.

Therefore this change does not fix the vulnerability of current
TA stack unwind process against core/TA isolation.

### CODE REVIEW

File: core/arch/arm/tee/arch_svc.c [hunk 1]
```diff
@@ -241,6 +242,9 @@ void __weak tee_svc_handler(struct thread_svc_regs *regs)
 	}
 }
 
+/*  Max size of the structure storing the  */
+#define TA_CONTEXT_MAX_SIZE		sizeof(struct thread_abort_regs)
```
Comment (jforissier) [inline core/arch/arm/tee/arch_svc.c]: Why not use the exact size of what is pushed by `utee_panic()`? That is: `14*4` for 32-bit TAs and `2*8` for 64-bit TAs.

File: core/arch/arm/tee/arch_svc.c [hunk 2]
```diff
@@ -274,6 +278,19 @@ static void print_panic_stack(struct thread_svc_regs *regs)
...
+
+	if (tee_ta_get_current_session(&s))
+		panic("No current session");
+
+	if (tee_mmu_check_access_rights(to_user_ta_ctx(s->ctx),
+					TEE_MEMORY_ACCESS_READ |
+					TEE_MEMORY_ACCESS_WRITE,
+					(uaddr_t)pregs,
+					TA_CONTEXT_MAX_SIZE)) {
+		EMSG("Invalid register dump, cannot print stack");
```
Comment (jforissier) [inline core/arch/arm/tee/arch_svc.c]: `TAMSG_RAW` for consistency.
Also, print pointer? `"Invalid user-mode pointer %0x" PRIxUA ",cannot print stack", pregs`

**Comment by jforissier** [comment]
@etienne-lms I don't think the "WHERE" patch is useful at all. If/when we decide to systematically check the accessibility of user space pointers during unwind, the TODO's won't help that much IMO.

**Comment by etienne-lms** [comment]
Oups, sorry. This patch was a private quick study i wanted to keep in mind. 
Pushed to the wrong branch :(  I will remove it.

**Comment by etienne-lms** [comment]
dummy change removed.
Comments NOT yet addressed.

**Comment by etienne-lms** [comment]
comments addressed, rebased.

File: core/arch/arm/tee/arch_svc.c [hunk 3]
```diff
@@ -241,6 +242,10 @@ void __weak tee_svc_handler(struct thread_svc_regs *regs)
 	}
 }
 
+/*  Max size of the structure storing the  */
```
Comment (jforissier) [inline core/arch/arm/tee/arch_svc.c]: Obsolete comment

**Comment by etienne-lms** [comment]
Comments addressed. 
Tag applied.

**Comment by jforissier** [comment]
CI is unhappy: missing cast to (uaddr_t) it seems.

**Comment by etienne-lms** [comment]
ok, thanks.

**Comment by etienne-lms** [comment]
rebased and squashed with tag.

**Pull request merged**